include/class_loader/multi_library_class_loader.h

```diff
@@ -100,15 +97,63 @@ class CLASS_LOADER_PUBLIC MultiLibraryClassLoader
   std::shared_ptr<Base> createInstance(const std::string & class_name, const std::string & library_path)
   {
     ClassLoader * loader = getClassLoaderForLibrary(library_path);
-    if (!loader) {
+    if (loader == NULL) {
       throw class_loader::NoClassLoaderExistsException(
-        "Could not create instance as there is no ClassLoader in MultiLibraryClassLoader bound to "
-        "library '" + library_path + "'. "
-        "Ensure you called MultiLibraryClassLoader::loadLibrary()");
+        "Could not create instance as there is no ClassLoader in "
+        "MultiLibraryClassLoader bound to library " + library_path +
+        " Ensure you called MultiLibraryClassLoader::loadLibrary()");
     }
     return loader->createInstance<Base>(class_name);
   }
 
+#if __cplusplus >= 201103L
+  /// Creates an instance of an object of given class name with ancestor class Base
+  /**
+   * This version does not look in a specific library for the factory, but rather the first open library that defines the classs
+   * @param Base - polymorphic type indicating base class
+   * @param class_name - the name of the concrete plugin class we want to instantiate
+   * @return A unique pointer to newly created plugin
+   */
+  template <class Base>
+  ClassLoader::UniquePtr<Base> createUniqueInstance(const std::string& class_name)
+  {
+    CONSOLE_BRIDGE_logDebug(
+      "class_loader::MultiLibraryClassLoader: Attempting to create instance of class type %s.",
+      class_name.c_str());
+    ClassLoader * loader = getClassLoaderForClass<Base>(class_name);
+    if (loader == nullptr) {
+      throw class_loader::CreateClassException(
+        "MultiLibraryClassLoader: Could not create object of class type " + class_name +
+        " as no factory exists for it. "
+        "Make sure that the library exists and was explicitly loaded through "
+        "MultiLibraryClassLoader::loadLibrary()");
+    }
+    return loader->createUniqueInstance<Base>(class_name);
+  }
+
+  /**
+   * @brief Creates an instance of an object of given class name with ancestor class Base
```

Comment:
Nit: all other functions in this PR have the brief outside of the commented block with `///`